### DISCUSSION TITLE
Chapter 1: Finding words, phrases... 11: Using the Matcher; incorrect use of matcher.add

### DISCUSSION THREAD

**Discussion by russhopper** [explosion/spaCy#7324] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour
The `matcher.add()` has `None` as the second param and this doesn't seem to adhere to documentation and doesn't run on my machine. 
https://spacy.io/api/matcher
```import spacy

# Import the Matcher
from spacy.matcher import Matcher

nlp = spacy.load("en_core_web_sm")
doc = nlp("Upcoming iPhone X release date leaked as Apple reveals pre-orders")

# Initialize the Matcher with the shared vocabulary
matcher = Matcher(nlp.vocab)

# Create a pattern matching two tokens: "iPhone" and "X"
pattern = [{"TEXT": "iPhone"}, {"TEXT": "X"}]

# Add the pattern to the matcher
matcher.add("IPHONE_X_PATTERN", None, pattern) ## THIS LINE

# Use the matcher on the doc
matches = matcher(doc)
print("Matches:", [doc[start:end].text for match_id, start, end in matches])
```

## Your Environment
* Operating System: Windows
* Python Version Used: Python 3
* spaCy Version Used: latest on Anaconda
* Environment Information: Using spaCy on Anaconda in a Jupyter notebook

**Comment by svlandeg** [comment]
Hi! This is a good question for the discussions forum, so I'll move it there. This issue will be closed, but replaced by a link to the new, open thread.

**Comment by svlandeg** [comment, marked as answer]
I'm afraid you may have run into an incompatibility issue between spaCy v2 and the newly released spaCy v3.

In short, you can remove the `None` argument and make sure you provide a list of patterns:
```
matcher.add("IPHONE_X_PATTERN", [pattern])
``` 

For more detailed information, see the docs here: https://spacy.io/api/matcher#add

 An updated version of the [spaCy course](https://course.spacy.io/) for the new spaCy v3 is coming soon as well.
